### ISSUE TITLE
Prettier plugins not found when VSCode launched outside of working directory

### ISSUE THREAD

**Issue by Sobes76rus** [prettier/prettier-vscode#2580]
### Summary

Prettier plugins not found when vscode was launched outside of working directory

### Github Repository to Reproduce Issue

https://github.com/Sobes76rus/prettier-vscode-bug

### Steps To Reproduce:

1. install any prettier plugin (in my case `@prettier/plugin-pug`) and launch `vscode` from `home` directory (for example)
2. open project folder in `vscode`
3. format any file, prettier will throw an error `cannot find module @prettier/plugin-pug`

### Expected result

no errors

### Actual result

error `cannot find module @prettier/plugin-pug`

### Additional information

if u launch vscode from project directory all will work fine

**VS Code Version**:
```
Version: 1.68.0
Commit: 4af164ea3a06f701fe3e89a2bcbb421d2026b68f
Date: 2022-06-08T11:49:57.055Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Linux x64 5.15.11-tkg-pds
```
**Prettier Extension Version**:

```
pnpm exec prettier -v                        
2.6.2
```

**OS and version**:
```
lsb_release -a                                                      
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 21.10
Release:	21.10
Codename:	impish
```
## Prettier Log Output

```
["INFO" - 1:03:07 AM] Formatting file:///home/artverum/Documents/Development/Work/contribution/prettier-vscode/package.json
["INFO" - 1:03:07 AM] Using config file at '/home/artverum/Documents/Development/Work/contribution/prettier-vscode/.prettierrc'
["INFO" - 1:03:07 AM] Using ignore file (if present) at /home/artverum/Documents/Development/Work/contribution/prettier-vscode/.prettierignore
["INFO" - 1:03:07 AM] File Info:
{
  "ignored": false,
  "inferredParser": "json-stringify"
}
["INFO" - 1:03:07 AM] Detected local configuration (i.e. .prettierrc or .editorconfig), VS Code configuration will not be used
["INFO" - 1:03:07 AM] Prettier Options:
{
  "filepath": "/home/artverum/Documents/Development/Work/contribution/prettier-vscode/package.json",
  "parser": "json-stringify",
  "plugins": [
    "@prettier/plugin-pug"
  ]
}
["ERROR" - 1:03:07 AM] Error formatting document.
["ERROR" - 1:03:07 AM] Cannot find module '@prettier/plugin-pug'
Require stack:
- /home/artverum/Documents/Development/Work/contribution/prettier-vscode/node_modules/.pnpm/prettier@2.6.2/node_modules/prettier/index.js
- /home/artverum/.vscode/extensions/esbenp.prettier-vscode-9.5.0/dist/extension.js
- /usr/share/code/resources/app/out/vs/loader.js
- /usr/share/code/resources/app/out/bootstrap-amd.js
- /usr/share/code/resources/app/out/bootstrap-fork.js
Error: Cannot find module '@prettier/plugin-pug'
Require stack:
- /home/artverum/Documents/Development/Work/contribution/prettier-vscode/node_modules/.pnpm/prettier@2.6.2/node_modules/prettier/index.js
- /home/artverum/.vscode/extensions/esbenp.prettier-vscode-9.5.0/dist/extension.js
- /usr/share/code/resources/app/out/vs/loader.js
- /usr/share/code/resources/app/out/bootstrap-amd.js
- /usr/share/code/resources/app/out/bootstrap-fork.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:987:15)
    at _.resolve (/usr/share/code/resources/app/out/vs/loader.js:4:761)
    at /home/artverum/Documents/Development/Work/contribution/prettier-vscode/node_modules/.pnpm/prettier@2.6.2/node_modules/prettier/index.js:39110:25
    at Array.map (<anonymous>)
    at Object.load (/home/artverum/Documents/Development/Work/contribution/prettier-vscode/node_modules/.pnpm/prettier@2.6.2/node_modules/prettier/index.js:39105:65)
    at Object.load [as loadPlugins] (/home/artverum/Documents/Development/Work/contribution/prettier-vscode/node_modules/.pnpm/prettier@2.6.2/node_modules/prettier/index.js:15998:27)
    at /home/artverum/Documents/Development/Work/contribution/prettier-vscode/node_modules/.pnpm/prettier@2.6.2/node_modules/prettier/index.js:39172:24
    at Object.Success [as format] (/home/artverum/Documents/Development/Work/contribution/prettier-vscode/node_modules/.pnpm/prettier@2.6.2/node_modules/prettier/index.js:39188:12)
    at t.default.format (/home/artverum/.vscode/extensions/esbenp.prettier-vscode-9.5.0/src/PrettierEditService.ts:463:45)
    at t.PrettierEditProvider.provideEdits (/home/artverum/.vscode/extensions/esbenp.prettier-vscode-9.5.0/src/PrettierEditService.ts:322:22)
    at M.provideDocumentFormattingEdits (/usr/share/code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:137239)
["INFO" - 1:03:07 AM] Formatting completed in 0.028ms.
```

**Comment by Eldair** [comment]
@ntotten 
This is definitely an issue but specific to pnpm (it works fine when installing via yarn)

**Comment by kftsehk** [comment]
# TLDR

Try restart vscode if plugin changed

# More information

I used `npm`, but it seem to me nothing to do with `npm`, my test only involve manually `rm` the plugin directories, there is definitely some caching mechanism, maybe with one of the parser, playing with me.

Problem is specific to `.js` file, for example, formatting `package.json` instantaneously reflect what change is done to `node_module/`.

Tested with
- RHEL8.7 on a laptop
- VSCode 1.76.0
- Prettier 9.10.4
- snippet of package.json / prettierrc at hand

package.json
```json
{
  "devDependencies": {
    "@babel/eslint-parser": "^7.19.1",
    "@babel/plugin-syntax-import-assertions": "^7.20.0",
    "@trivago/prettier-plugin-sort-imports": "^4.1.1",
    "depcheck": "^1.4.3",
    "eslint": "^8.35.0",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-config-prettier": "^8.6.0",
    "eslint-plugin-import": "^2.27.5",
    "eslint-plugin-json": "^3.1.0",
    "eslint-plugin-promise": "^6.1.1",
    "eslint-plugin-security": "^1.7.1",
    "prettier": "^2.8.4",
    "prettier-plugin-packagejson": "^2.4.3",
    "prettier-plugin-sort-json": "^1.0.0"
  }
}
```

prettierrc.yml
```yaml
bracketSameLine: false
importOrder:
  - '^@.*$'
  - '^[a-z].*$'
  - '^\.\.?/.*\.mjs$'
  - '^\.\.?/.*\.json$'
importOrderSeparation: true
importOrderSortSpecifiers: true
jsonRecursiveSort: true
overrides:
  - files:
      - '*.yml'
      - '*.yaml'
    options:
      bracketSpacing: false
      printWidth: 10000
plugins:
  - 'prettier-plugin-sort-json'
  - 'prettier-plugin-packagejson'
semi: true
singleQuote: true
tabWidth: 2
trailingComma: 'es5'
```

## Behavior

```bash
# create the above files
npm i
touch index.js
code .
# configure prettier as js formatter, modify index.js, try to format everything is file (expected)
```

### Strange behavior 1:
```bash
rm node_module/prettier-plugin-* -rf
```
now prettier is still working for `index.js` even the plugin directories are removed (why? shouldn't work?), but
```bash
cp index.js index2.js
```
prettier does not work on index2.js (expected)

---

```bash
# with node_module/prettier-plugin-* removed, close vscode
code .
# expect prettier not work (expected)
```

### Strange behavior 2
```bash
npm i
```
Now even with all plugin restored, prettier does not work for every js file, including new files subsequently created and opened after `npm i` is done (Is this the behavior the original issue is referring to?)

---

```bash
# close vscode
code .
# works again (expected)
```

**Comment by indirectlylit** [comment]
Thank you!

This was driving me nuts. I had the same issue with `require('@tailwindcss/aspect-ratio')` inside the tailwind config, but it probably had nothing to do with that...

> Try restart vscode if plugin changed

It also works to just run `Developer: Reload Window`

**Comment by kftsehk** [comment]
I think this has to do with caching the linter instance? could it be a plugin issue or vscode's own issue
